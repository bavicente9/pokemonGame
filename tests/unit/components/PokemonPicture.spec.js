import PokemonPicture from '@/components/PokemonPicture'
import {shallowMount} from '@vue/test-utils'

describe("PokemonPicture Component", () => {
  
    test('match with snapshot', () => {
      
        const wrapper = shallowMount(PokemonPicture, {
            props:{
                pokemonId:1,
                showPokemon:false
            }
        })
        
        expect(wrapper.html()).toMatchSnapshot()
    })
    
    test('show image, pokemon n100',() => {
        const wrapper = shallowMount(PokemonPicture, {
            props:{
                pokemonId:100,
                showPokemon:false
            }
        })
        
        const [img1,img2] = wrapper.findAll('img')
        
        expect(img1.exists()).toBe(true)
        expect(img2).toBe(undefined)
        
        expect(img1.classes('hidden-pokemon')).toBe(true)
        expect(img1.attributes().src).toBe(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg`)
        
    })
    
    test('show pokemon if showPokemon:true', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props:{
                pokemonId:100,
                showPokemon:true
            }
        })
        const img1 = wrapper.find('img')
        expect(img1.exists()).toBe(true)
        
        expect(img1.classes('hidden-pokemon')).toBe(false)
      
    })

})