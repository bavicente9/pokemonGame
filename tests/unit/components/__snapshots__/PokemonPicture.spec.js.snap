// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture Component match with snapshoot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="Pokemon" class="hidden-pokemon"></div>`;
